test/parallel/test-fs-stat-bigint.js

```diff
@@ -0,0 +1,145 @@
+        numStats.isBlockDevice()
+      );
+      assert.strictEqual(
+        bigintStats.isCharacterDevice(),
+        numStats.isCharacterDevice()
+      );
+      assert.strictEqual(
+        bigintStats.isDirectory(),
+        numStats.isDirectory()
+      );
+      assert.strictEqual(
+        bigintStats.isFIFO(),
+        numStats.isFIFO()
+      );
+      assert.strictEqual(
+        bigintStats.isFile(),
+        numStats.isFile()
+      );
+      assert.strictEqual(
+        bigintStats.isSocket(),
+        numStats.isSocket()
+      );
+      assert.strictEqual(
+        bigintStats.isSymbolicLink(),
+        numStats.isSymbolicLink()
+      );
+    } else if (Number.isSafeInteger(val)) {
+      // eslint-disable-next-line no-undef
+      assert.strictEqual(bigintStats[key], BigInt(val));
+    } else {
+      assert(
+        Math.abs(Number(bigintStats[key]) - val) < 1,
+        `${key} is not a safe integer, difference should < 1.\n` +
+        `Number version ${val}, BigInt version ${bigintStats[key]}n`);
+    }
+  }
+}
+
+{
+  const bigintStats = fs.statSync(fn, { bigint: true });
+  const numStats = fs.statSync(fn);
+  verifyStats(bigintStats, numStats);
+}
+
+{
+  const bigintStats = fs.lstatSync(link, { bigint: true });
+  const numStats = fs.lstatSync(link);
+  verifyStats(bigintStats, numStats);
+}
+
+{
+  const fd = fs.openSync(fn, 'r');
+  const bigintStats = fs.fstatSync(fd, { bigint: true });
+  const numStats = fs.fstatSync(fd);
+  verifyStats(bigintStats, numStats);
+  fs.closeSync(fd);
+}
+
+{
+  fs.stat(fn, { bigint: true }, (err, bigintStats) => {
+    fs.stat(fn, (err, numStats) => {
+      verifyStats(bigintStats, numStats);
+    });
+  });
+}
+
+{
+  fs.lstat(link, { bigint: true }, (err, bigintStats) => {
+    fs.lstat(link, (err, numStats) => {
+      verifyStats(bigintStats, numStats);
+    });
+  });
+}
+
+{
+  const fd = fs.openSync(fn, 'r');
+  fs.fstat(fd, { bigint: true }, (err, bigintStats) => {
+    fs.fstat(fd, (err, numStats) => {
+      verifyStats(bigintStats, numStats);
+      fs.closeSync(fd);
+    });
+  });
+}
+
+(async function() {
+  const bigintStats = await promiseFs.stat(fn, { bigint: true });
+  const numStats = await promiseFs.stat(fn);
+  verifyStats(bigintStats, numStats);
+})();
+
+(async function() {
+  const bigintStats = await promiseFs.lstat(link, { bigint: true });
+  const numStats = await promiseFs.lstat(link);
+  verifyStats(bigintStats, numStats);
+})();
+
+(async function() {
+  const handle = await promiseFs.open(fn, 'r');
+  const bigintStats = await promiseFs.fstat(handle, { bigint: true });
+  const numStats = await promiseFs.fstat(handle);
+  verifyStats(bigintStats, numStats);
+  await handle.close();
+})();
+
+(async function() {
+  const handle = await promiseFs.open(fn, 'r');
+  const bigintStats = await handle.stat({ bigint: true });
+  const numStats = await handle.stat();
+  verifyStats(bigintStats, numStats);
+  await handle.close();
+})();
```

Comment:
These tests would all pass in case e.g. `const handle = await promiseFs.open(fn, 'r');` rejects. In that case it would log an unhandled rejection and that would currently not end the process, so we would not notice anything wrong.

Instead, it would be good to write something like:

```js
async function foo() {
}

Promise.all([
  foo(),
  bar(),
  ...
]).then(common.mustCall());

Reply:
There is also a `common` function to throw in case of unhandled rejections.

Reply:
I went with `common.crashOnUnhandledRejection()` because it's simple